### PR TITLE
Pass through user kwargs to rasterio.features.rasterize

### PR DESCRIPTION
**Pull request by darothen**
This is a simple code change that lets you pass kwargs through to [`rasterio.features.rasterize`](https://mapbox.github.io/rasterio/api/rasterio.features.html#rasterio.features.rasterize). In particular, I wanted to use the  **all_touched** kwarg to include cells clipped by my shapefile/regions.

A simple example using the test ozone data I previously posted an issue about. By default, this is the sort of plot we get, using a custom function which plots our map:

``` python
us_subset = ozone_subset.salem.roi(shape=usa)
x = us_subset.where(~us_subset.isnull(), drop=True)
ax = plot_map(x)

import numpy as np
xx, yy = np.meshgrid(x.lon, x.lat)
ax.scatter(xx, yy, marker='x', color='k', s=3)
```

image

Notice how the coasts are clipped and we miss grid cells covering them, particularly Florida, Texas, and California.

Repeating, but passing the **all_touched** kwarg:

``` python
us_subset = ozone_subset.salem.roi(shape=usa, rasterize_kws=dict(all_touched=True))
x = us_subset.where(~us_subset.isnull(), drop=True)
ax = plot_map(x)

import numpy as np
xx, yy = np.meshgrid(x.lon, x.lat)
ax.scatter(xx, yy, marker='x', color='k', s=3)
```

image

[`rasterize`](https://mapbox.github.io/rasterio/api/rasterio.features.html#rasterio.features.rasterize) doesn't really expose too many arguments, so this may be superfluous - it might be simpler just to directly add the **all_touched** argument inside `Grid.region_of_interest`,

### CODE REVIEW

**Comment by fmaussion** [comment]
This is great, thanks! I think none of the other kwargs are useful for a ROI so I'd be fine with an ``all_touched`` kwarg, but I leave it up to you.

Also, could you PR to master instead? (don't know if this is possible without restarting a PR)

File: salem/gis.py [hunk 1]
```diff
@@ -938,7 +946,8 @@ def region_of_interest(self, shape=None, geometry=None, grid=None,
                                         inplace=inplace)
             import rasterio
             with rasterio.Env():
-                mask = rasterio.features.rasterize(shape.geometry, out=mask)
+                mask = rasterio.features.rasterize(shape.geometry,
+                                                   **rasterize_kws)
         if geometry is not None:
```
Comment (fmaussion) [inline salem/gis.py]: Could you also add the kwarg to the rasterize call in the ``if geometry is not None`` block just below?

**Comment by darothen** [comment]
Will do. Here's the to-do list I'll tackle right now.

- [ ] Re-targeted to master
- [ ] Add rasterize call in `if geometry is not None` block
- [ ] Re-factor to just **all_touched**

**Comment by darothen** [comment]
Updated... let me know if there's anything else you'd like me to tweak!

**Comment by fmaussion** [comment]
perfect, Thanks @darothen !

**Pull request merged**